### DISCUSSION TITLE
Display triangle header with index information and handling manual LDF selection

### DISCUSSION THREAD

**Discussion by vdunaway** [casact/chainladder-python#495] [General]
My question is twofold:

1. I'm trying to loop through multiple triangles and output to excel on different sheets for the user to have an easier look at the output and various averages, compare to a benchmark/industry, and ultimately make a final selection. I want to be able to output the labels of what that triangle is displaying based on its indices. For example, maybe the triangle is: **Auto Liability, non-cat, untrended, Cumulative Incurred triangle**, it would be helpful to have this output with the triangle. This makes it easier to know exactly what information is being displayed in the triangle. In addition, outputting the name of the origin as "Accident Year". As you can see in the example below, none of that important information is displayed so a user wouldn't really know what iloc[0,0] is or what the triangle is actually displaying.

![Screenshot 2024-03-15 083332](https://github.com/casact/chainladder-python/assets/159925326/0ef47c22-86f6-4dae-b281-fcf45103a930)

2. If I select an LDF pattern in excel, how do I properly feed that back to chainladder to apply to a triangle?

**Comment by jbogaardt** [comment]
Hi @vdunaway, Great questions! The heatmap isn't currently designed to be customizable. I think we can redesign it so that it perhaps returns a pandas.style object. From there, one should be able to further customize it - including adding a title. This should be flagged as an ehnacement.

If you have patterns that you've developed externally, then you can include those using the `DevelopmentConstant` estimator. Refer to this [example](https://chainladder-python.readthedocs.io/en/latest/user_guide/development.html#developmentconstant) from the docs. This package isn't opinionated on how you make Excel and Python work together, but if you can force your excel inputs into the dictionary format from the example, you should be able to incorporate external patterns. For a more complex use-case of including multiple patterns in  one fell-swoop, refer to this [example](https://chainladder-python.readthedocs.io/en/latest/gallery/plot_callable_dev_constant.html).

**Reply by vdunaway** [reply]
This is perfect for my second question! The examples are very helpful.

To further expand on my first question, which I think that enhancement would be great, what I haven't been able to figure out is even at a basic information output level (not nicely formatted title), is it possible to print/output the index names of a particular triangle? If I wanted to get the index names in my example of `tri.iloc[0,0]`, is that possible?

**Comment by jbogaardt** [comment]
Not elaborate, but you can just print the index and columns of the referenced slice:
```python
print(triangle.iloc[0, 0].index, triangle.iloc[0, 0].columns)
```
